### ISSUE TITLE
 training generated weights to test has no result

### ISSUE THREAD

**Issue by Aiz-wallenstein** [zylo117/Yet-Another-EfficientDet-Pytorch#414]
I  trained 20 images from coco dataset to test, and it generated some weights, but when I use these weights to run efficientdet_test.py, it  has no result. What causes this happen?

**Comment by zylo117** [comment]
did you train with pretrained weights?

**Comment by Aiz-wallenstein** [comment]
I use your train code without changes.

**Comment by zylo117** [comment]
no, I mean did you train with pretrained weights?

**Comment by Aiz-wallenstein** [comment]
is this? I think I didn't use pretrained weight. 
`  parser = argparse.ArgumentParser('Yet Another EfficientDet Pytorch: SOTA object detection network - Zylo117')
    parser.add_argument('-p', '--project', type=str, default='coco', help='project file that contains parameters')
    parser.add_argument('-c', '--compound_coef', type=int, default=0, help='coefficients of efficientdet')
    parser.add_argument('-n', '--num_workers', type=int, default=0, help='num_workers of dataloader')
    parser.add_argument('--batch_size', type=int, default=4, help='The number of images per batch among all devices')
    parser.add_argument('--head_only', type=boolean_string, default=False,
                        help='whether finetunes only the regressor and the classifier, '
                             'useful in early stage convergence or small/easy dataset')
    parser.add_argument('--lr', type=float, default=1e-4)
    parser.add_argument('--optim', type=str, default='adamw', help='select optimizer for training, '
                                                                   'suggest using \'admaw\' until the'
                                                                   ' very final stage then switch to \'sgd\'')
    parser.add_argument('--num_epochs', type=int, default=50)
    parser.add_argument('--val_interval', type=int, default=1, help='Number of epoches between valing phases')
    parser.add_argument('--save_interval', type=int, default=10, help='Number of steps between saving')
    parser.add_argument('--es_min_delta', type=float, default=0.0,
                        help='Early stopping\'s parameter: minimum change loss to qualify as an improvement')
    parser.add_argument('--es_patience', type=int, default=0,
                        help='Early stopping\'s parameter: number of epochs with no improvement after which training will be stopped. Set to 0 to disable this technique.')
    parser.add_argument('--data_path', type=str, default='F:', help='the root folder of dataset')
    parser.add_argument('--log_path', type=str, default='logs/')
    parser.add_argument('-w', '--load_weights', type=str, default=None,
                        help='whether to load weights from a checkpoint, set None to initialize, set \'last\' to load last checkpoint')
    parser.add_argument('--saved_path', type=str, default='logs/')
    parser.add_argument('--debug', type=boolean_string, default=False, help='whether visualize the predicted boxes of training, '
                                                                  'the output images will be in test/')
`

**Comment by zylo117** [comment]
No. 
And you should.
You shouldn't expect effdet can be trained on 20 images

**Comment by Aiz-wallenstein** [comment]
Thanks! I have understood this problem. And I want to know how to load the pretrained weight, I see the readme and tried, but it 
didn't seem to work.

**Comment by zylo117** [comment]
Does training with `args -w path/to/weights` work?

**Comment by Aiz-wallenstein** [comment]
Do you mean to write the path of pre training weight here？This seems to be using this weight as a checkpoint.
` parser.add_argument('-w', '--load_weights', type=str, default=None,
                        help='whether to load weights from a checkpoint, set None to initialize, set \'last\' to load last checkpoint')`

And in the efficietnet/utils.py line 299, 
`def load_pretrained_weights(model, model_name, load_fc=True, advprop=False):`
what does this function means? Isn't this the function used to load the pre training weights？

**Comment by mp-lu** [comment]
if you set -c 0 , then try to add "--load_weights weights/efficientdet-d0.pth" in your command.
The pretrained weights can download at the  readme.md.
